File: src/main/java/Chapter02/color/ColorMain.java
package Chapter02.color;

public class ColorMain {

    public static void main(String[] args) {
        System.out.println(getMood(ColorSet.RED));
    }

    public static String getMood(ColorSet color) {
        String result = "";
        switch (color) {
            case RED:
            case ORANGE:
            case YELLOW:
                result = "warm";
                break;
            case GREEN:
                result = "neutral";
                break;
            case BLUE:
            case INDIGO:
            case VIOLET:
                result = "cold";
                break;
        }
        return result;
    }

}

File: src/main/kotlin/Chapter02/color/ColorMain.kt
package Chapter02.color

import Chapter02.color.Color.*

fun getMood(color: Color) =
    when (color) {
        RED, ORANGE, YELLOW -> "warm"
        GREEN -> "neutral"
        BLUE, INDIGO, VIOLET -> "cold"
    }

fun mix(color1: Color, color2: Color) =
    when (setOf(color1, color2)) {
        setOf(RED, YELLOW) -> ORANGE
        setOf(YELLOW, BLUE) -> GREEN
        setOf(BLUE, VIOLET) -> INDIGO
        else -> throw Exception("Dirty color")
    }

fun main() {
    println(getMood(RED))
    println(mix(RED, YELLOW))
}